tests/dependencies/is_narwhals_dataframe_test.py

```diff
@@ -0,0 +1,15 @@
+from __future__ import annotations
+
+import pandas as pd
+
+import narwhals as nw
+import narwhals.stable.v1 as nws
+from narwhals.stable.v1.dependencies import is_narwhals_dataframe
+
+
+def test_is_narwhals_dataframe() -> None:
+    df = pd.DataFrame({"col1": [1, 2], "col2": [3, 4]})
+
+    assert is_narwhals_dataframe(nw.from_native(df))
+    assert is_narwhals_dataframe(nws.from_native(df))
+    assert not is_narwhals_dataframe(df)
```

Comment:
Instead of importing and testing only pandas, you can use the the `constructor_eager` fixture as:

```suggestion
data = {"col1": [1, 2], "col2": [3, 4]}

def test_is_narwhals_dataframe(constructor_eager) -> None:
    df_native = constructor_eager(data)
    df_nw = nw.from_native(df_native, eager_only=True)
    df_nws = nws.from_native(df_native, eager_only=True)
    
    assert is_narwhals_dataframe(df_nw)
    assert is_narwhals_dataframe(df_nws)
    assert not is_narwhals_dataframe(df_native)
```

to test against all possible dataframe constructors